src/token/history/history.service.ts

```diff
@@ -0,0 +1,65 @@
+import { Injectable, NotFoundException } from '@nestjs/common'
+import { Prisma } from '@prisma/client'
+import { PrismaService } from '../../database/prisma.service'
+import { TokenHistory } from '../../token/history/entities/token-history.entity'
+
+@Injectable()
+export class HistoryTokenService {
+    constructor(private prisma: PrismaService) {}
+
+    async getMyRoomById(id: number): Promise<TokenHistory> {
+        const token = await this.prisma.tokenHistory.findUnique({
+            where: {
+                id,
+            },
+        })
+
+        if (!token) {
+            throw new NotFoundException('token not found')
+        }
+
+        return token
+    }
+
+    async createMyRoom(data: Prisma.TokenHistoryCreateInput): Promise<TokenHistory> {
+        const tokens = await this.prisma.tokenHistory.create({ data })
+
+        return tokens
+    }
+
+    async updateMyRoom(id: number, data: Prisma.TokenHistoryUpdateInput): Promise<TokenHistory> {
+        const token = await this.prisma.tokenHistory.findUnique({
```

Comment:
No need to check if exist object before update it. We do it in resolver only and throw exeption Error